var/spack/repos/builtin/packages/intel-mkl/package.py

```diff
@@ -96,19 +96,19 @@ def lapack_libs(self):
     @property
     def scalapack_libs(self):
         libnames = ['libmkl_scalapack']
-        if self.spec.satisfies('^openmpi'):
+        root = self.spec.root
+        if '^openmpi' in root:
             libnames.append('libmkl_blacs_openmpi')
-        elif self.spec.satisfies('^mpich@1'):
+        elif '^mpich@1' in root:
             libnames.append('libmkl_blacs')
-        elif self.spec.satisfies('^mpich@2:'):
+        elif '^mpich@2:' in root:
             libnames.append('libmkl_blacs_intelmpi')
-        elif self.spec.satisfies('^mvapich2'):
+        elif '^mvapich2' in root:
             libnames.append('libmkl_blacs_intelmpi')
-        elif self.spec.satisfies('^mpt'):
+        elif '^mpt' in root:
             libnames.append('libmkl_blacs_sgimpt')
-        # TODO: ^intel-parallel-studio can mean intel mpi, a compiler or a lib
-        # elif self.spec.satisfies('^intel-parallel-studio'):
-        #     libnames.append('libmkl_blacs_intelmpi')
+        elif 'intel-mpi' in root:
```

Comment:
I'll change it as it is obviously a typo. Side note: due to the semantic explained in #4135 this was working. I assume that:
```python
Spec('intel-mpi').satisfies('intel-mpi')
```
was tested when traversing the DAG.